code/modules/overmap/ftl_shunt/computer.dm

```diff
@@ -23,6 +28,57 @@
 	var/jump_cost = ((linked.vessel_mass * JOULES_PER_TON) / 1000) * jump_dist
 	return jump_cost
 
+/obj/machinery/computer/ship/ftl/proc/recalc_cost_power()
+	if(!linked_core)
+		return INFINITY
+	var/jump_dist = get_dist(linked, locate(linked_core.shunt_x, linked_core.shunt_y, global.using_map.overmap_z))
+	var/jump_cost = ((linked.vessel_mass * JOULES_PER_TON) / 1000) * jump_dist
+	var/jump_cost_power = jump_cost * REQUIRED_CHARGE_MULTIPLIER
+	return jump_cost_power
+
+/obj/machinery/computer/ship/ftl/proc/get_status()
+	if(is_jump_unsafe())
+		return JUMP_STATUS_DANGEROUS
+	if(plotting_jump)
+		return JUMP_STATUS_PLOTTING
+	if(jump_plotted == FALSE)
+		return JUMP_STATUS_NO_PLOT
+	else
+		return JUMP_STATUS_SAFE
+
+/obj/machinery/computer/ship/ftl/proc/is_jump_unsafe()
+	. = FALSE
+	var/dist = get_dist(locate(linked_core.shunt_x, linked_core.shunt_y, global.using_map.overmap_z), get_turf(linked))
+	if(dist > linked_core.safe_jump_distance)
+		. = TRUE
+
+/obj/machinery/computer/ship/ftl/proc/plot_jump(var/x, var/y)
+	var/jump_dist = get_dist(linked, locate(x, y, global.using_map.overmap_z))
+	var/plot_delay_mult
+	var/delay
+	switch(jump_dist)
+		if(1 to linked_core.safe_jump_distance)
+			plot_delay_mult = 1
+		if(linked_core.safe_jump_distance to linked_core.moderate_jump_distance)
+			plot_delay_mult = 1.5
+		if(linked_core.moderate_jump_distance to INFINITY)
+			plot_delay_mult = 2
+
+	delay = (jump_dist * BASE_PLOT_TIME_PER_TILE) * plot_delay_mult
```

Comment:
It's possible this is 0 or something else due to how `get_dist` actually works. Should probably clamp either jump_dist or this.